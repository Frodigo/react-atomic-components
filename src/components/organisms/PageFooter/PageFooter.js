import React from 'react';
import PropTypes from 'prop-types';

import FooterColumn from '../../molecues/FooterColumn';
import { Link } from '../../atoms';

import classes from './PageFooter.module.css';

const PageFooter = (props) => {
    const { footerColumns, footerBottomLinks, copyright, ...rest } = props;

    const shouldDisplayFooterColumns = footerColumns.map((col) => {
        return (
            <FooterColumn
                heading={col.header}
                subheadings={col.links}
                key={col.header}
            />
        );
    });

    const shouldDisplayFooterBottomLinks = footerBottomLinks.map((link) => {
        return (
            <li key={link.label}>
                <Link href={link.href} >{link.label}</Link>
            </li>
        );
    });

    return (
        <footer className={classes.pageFooter} {...rest}>
            <div className={classes.columns}>{shouldDisplayFooterColumns}</div>

            <div>
                <ul className={classes.bottomNav}>
                    {shouldDisplayFooterBottomLinks}
                </ul>
                <p className={classes.copyright}>
                    {copyright}
                </p>
            </div>
        </footer>
    );
};

PageFooter.propTypes = {
    footerColumns: PropTypes.arrayOf(
        PropTypes.shape({
            header: PropTypes.string,
            links: PropTypes.array
        })
    )
};

export default PageFooter;
