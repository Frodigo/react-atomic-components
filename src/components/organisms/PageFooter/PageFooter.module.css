.pageFooter {
    background-color: var(--color-alternate);
    color: var(--color-gray-light);
}

.bottomNav {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    margin-bottom: 0;
    padding: 10px;
}

.bottomNav li {
    margin: 0;
}

.bottomNav li::before {
    display: none;
}

.bottomNav a {
    color: inherit;
    display: inline-flex;
    font-size: 14px;
    font-weight: 400;
    padding: 10px;
}

.copyright {
    border-top: 1px solid var(--color-secondary);
    text-align: center;
    padding: var(--spacing-xl) 0 70px;
}

@media (min-width: 1024px) {  
    .columns {
        display: flex;
        flex-wrap: wrap;
    }

    .copyright {
        padding: var(--spacing-xl);
    }
}

@media (min-width: 1300px) {  
    .columns,
    .copyright {
        max-width: var(--desktop-container-width);
        margin: 0 auto;
    }

    .bottomNav a:hover,
    .bottomNav a:focus {
        text-decoration: none;
    }
}
