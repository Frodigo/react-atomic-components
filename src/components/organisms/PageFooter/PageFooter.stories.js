import React from 'react';

import { default as PageFooterComponent } from './PageFooter';

export default {
    title: 'Organisms',
    component: PageFooterComponent
};

const Template = (args) => <PageFooterComponent {...args} />;

export const PageFooter = Template.bind({});
