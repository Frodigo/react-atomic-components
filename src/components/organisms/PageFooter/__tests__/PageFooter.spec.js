import React from 'react';
import { render } from '@testing-library/react';

import PageFooter from '../PageFooter';

describe('Page footer should be rendered', () => {
    test('Page footer renders correctly', () => {
        const { container } = render(<PageFooter footerColumns={[
            {
                header: 'Column 1',
                links: [{
                    url: '#',
                    label: 'lorem ipsum'
                },{
                    url: '#',
                    label: 'dolor sit amet'
                }]
            },
            {
                header: 'Column 2',
                links: [{
                    url: '#',
                    label: 'lorem ipsum'
                },{
                    url: '#',
                    label: 'dolor sit amet'
                }]
            },
            {
                header: 'Column 3',
                links: [{
                    url: '#',
                    label: 'lorem ipsum'
                },{
                    url: '#',
                    label: 'dolor sit amet'
                }]
            },
            {
                header: 'Column 4',
                links: [{
                    url: '#',
                    label: 'lorem ipsum'
                },{
                    url: '#',
                    label: 'dolor sit amet'
                }]
            }
        ]} footerBottomLinks={[
            {
                href: '#',
                label: 'Link 1'
            },
            {
                href: '#',
                label: 'Link 2'
            },
            {
                href: '#',
                label: 'Link 3'
            }
        ]} copyright="&copy; 2021 Lorem ipsum" />);

        expect(container).toMatchSnapshot();
    });
});
