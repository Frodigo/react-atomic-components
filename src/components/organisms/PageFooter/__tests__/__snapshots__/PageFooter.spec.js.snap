// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page footer should be rendered Page footer renders correctly 1`] = `
<div>
  <footer
    class="pageFooter"
  >
    <div
      class="columns"
    >
      <div
        class="footerColumn"
      >
        <button
          class="header"
          type="button"
        >
          <h5
            class="heading"
          >
            Column 1
          </h5>
          <span
            class="expandButton"
          >
            <span>
              svg
            </span>
          </span>
        </button>
      </div>
      <div
        class="footerColumn"
      >
        <button
          class="header"
          type="button"
        >
          <h5
            class="heading"
          >
            Column 2
          </h5>
          <span
            class="expandButton"
          >
            <span>
              svg
            </span>
          </span>
        </button>
      </div>
      <div
        class="footerColumn"
      >
        <button
          class="header"
          type="button"
        >
          <h5
            class="heading"
          >
            Column 3
          </h5>
          <span
            class="expandButton"
          >
            <span>
              svg
            </span>
          </span>
        </button>
      </div>
      <div
        class="footerColumn"
      >
        <button
          class="header"
          type="button"
        >
          <h5
            class="heading"
          >
            Column 4
          </h5>
          <span
            class="expandButton"
          >
            <span>
              svg
            </span>
          </span>
        </button>
      </div>
    </div>
    <div>
      <ul
        class="bottomNav"
      >
        <li>
          <a
            href="#"
          >
            Link 1
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Link 2
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Link 3
          </a>
        </li>
      </ul>
      <p
        class="copyright"
      >
        Â© 2021 Lorem ipsum
      </p>
    </div>
  </footer>
</div>
`;
